git_tools/create_opence_release.py

```diff
@@ -0,0 +1,114 @@
+    parser = utils.make_parser([git_utils.Argument.PUBLIC_ACCESS_TOKEN, git_utils.Argument.REPO_DIR,
+                                git_utils.Argument.BRANCH],
+                               description = 'A script that can be used to cut an open-ce release.')
+
+    parser.add_argument(
+        '--github-org',
+        type=str,
+        default="open-ce",
+        help="""Org to cut an Open-CE release in.""")
+
+    parser.add_argument(
+        '--primary-repo',
+        type=str,
+        default="open-ce",
+        help="""Primary open-ce repo.""")
+
+    parser.add_argument(
+        '--version',
+        type=str,
+        required=True,
+        help="""Release version to cut.""")
+
+    return parser
+
+def _main(arg_strings=None):
+    parser = _make_parser()
+    args = parser.parse_args(arg_strings)
+    version_name = "open-ce-v{}".format(args.version)
+    release_number = ".".join(args.version.split(".")[:-1])
+    branch_name = "open-ce-r{}".format(release_number)
+    version_msg = "Open-CE Version {}".format(args.version)
+    primary_repo_url = "git@github.com:{}/{}.git".format(args.github_org, args.primary_repo)
+
+    primary_repo_path = os.path.abspath(os.path.join(args.repo_dir, args.primary_repo))
+    print("--->Making clone location: " + primary_repo_path)
+    os.makedirs(primary_repo_path, exist_ok=True)
+    print("--->Cloning {}".format(primary_repo_url))
+    git_utils.clone_repo(primary_repo_url, primary_repo_path, args.branch)
+
+    print("--->Creating {} branch in {}".format(version_name, args.primary_repo))
+    git_utils.create_branch(primary_repo_path, branch_name)
+
+    print("--->Updating env files.")
+    _update_env_files(primary_repo_path, version_name)
+
+    print("--->Committing env files.")
+    git_utils.commit_changes(primary_repo_path, "Updates for {}".format(release_number))
+
+    print("--->Tag Primary Branch")
+    git_utils.create_tag(primary_repo_path, version_name, version_msg)
+
+    push = git_utils.ask_for_input("Would you like to push changes to primary repo?")
+    if push.startswith("y"):
+        print("--->Pushing branch.")
+        git_utils.push_branch(primary_repo_path, branch_name)
+        print("--->Pushing tag.")
+        git_utils.push_branch(primary_repo_path, version_name)
+
+    tag_all_repos.tag_all_repos(github_org=args.github_org,
+                                tag=version_name,
+                                tag_msg=version_msg,
+                                branch=args.branch,
+                                repo_dir=args.repo_dir,
+                                pat=args.pat,
+                                skipped_repos=args.primary_repo)
+
+def _update_env_files(open_ce_path, new_git_tag):
+    for env_file in glob.glob(os.path.join(open_ce_path, "envs", "*.yaml")):
+        print("--->Updating {}".format(env_file))
+        with open(env_file, 'r') as content_file:
+            env_file_contents = content_file.read()
```

Comment:
Should this be loaded with pyyaml's safe_load function?

Reply:
That wouldn't work with environment file that have jinja would it?
I don't actually want to parse the yaml, I just want to treat it as text.

Reply:
Ah, that is a good point.